build.py

```diff
@@ -0,0 +1,68 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPL-3.0 2016-present Scille SAS
+
+import os
+import sys
+import logging
+import subprocess
+
+YELLOW_FG = "\x1b[33m"
+DEFAULT_FG = "\x1b[39m"
+
+log = logging.getLogger("build")
+logging.basicConfig(
+    level=logging.DEBUG, format=f"{YELLOW_FG}[%(levelname)s] %(message)s{DEFAULT_FG}"
+)
+
+PYTHON_PATH = sys.executable
+log.debug(f"PYTHON_PATH={PYTHON_PATH}")
+
+
+def run(cmd, **kwargs):
+    log.debug(f">>> {cmd}")
+    ret = subprocess.run(cmd, shell=True, check=True, **kwargs)
+    return ret
+
+
+def in_cibuildwheel():
+    return bool(int(os.environ.get("CIBUILDWHEEL", "0")))
+
+
+def check_venv():
+    # Both conda and venv are configured, pick conda
+    if os.environ.get("CONDA_PREFIX") and os.environ.get("VIRTUAL_ENV"):
+        del os.environ["VIRTUAL_ENV"]
+    # Build with cibuildwheel detected, set `VIRTUAL_ENV`
+    if in_cibuildwheel() and os.environ.get("VIRTUAL_ENV") is None:
+        os.environ["VIRTUAL_ENV"] = os.path.dirname(os.path.dirname(PYTHON_PATH))
+        log.debug(
+            f"Build with cibuildwheel detected, set `VIRTUAL_ENV` as: {os.environ['VIRTUAL_ENV']}"
+        )
+    if not os.environ.get("CONDA_PREFIX") and not os.environ.get("VIRTUAL_ENV"):
+        raise RuntimeError("A virtual env is required to run `maturin develop`")
+
+
+def force_maturin_release() -> bool:
+    return os.environ.get("FORCE_MATURIN_RELEASE", "0") == "1"
+
+
+def build():
+    # Debug
+    log.debug("Environment:")
```

Comment:
Do we have a reason for dump the environment variables like this ? (I have the feeling this is some debug code useful we creating this script, but which should be removed now that the code is stable)

This might not be a good idea for instance in the CI if we expose a github secret as environment variable...

Reply:
done